### DISCUSSION TITLE
内容忽略正则表达式集合

### DISCUSSION THREAD

**Discussion by YiiGuxing** [YiiGuxing/TranslationPlugin#1209] [Q&A] [answered]
# 内容忽略正则表达式集合

**内容忽略**功能使用正则表达式进行匹配，使用**内容忽略**功能可以忽略掉你不想送去翻译的内容，比如注释和文档中的干扰内容（如Java中的 `/` 和 `*` ）：

![image](https://user-images.githubusercontent.com/10737066/147811466-9c82c966-4798-4992-8d80-f29c6dc5886c.png)

下面分享几个可能有用的内容忽略正则表达式，也欢迎大家补充更多、更好、功能更强大、支持更多语言的内容忽略正则表达式。

**Comment by YiiGuxing** [comment, marked as answer]
```regexp
[/*]+
```
  > **`Java`, `JavaScript` 等语言的注释内容忽略正则表达式**
  >
  > *最简单粗暴，缺点是注释和文档内容中的 `/` 和 `*` 也会被忽略掉，效果如下图所示：*
  >
  > ![1](https://user-images.githubusercontent.com/10737066/67548287-0a9e0900-f734-11e9-9ddc-143e0df63e96.png)

**Comment by YiiGuxing** [comment]
```regexp
(^\s*(/(/+|\*+)|(\*+(?!/))) ?)|(\s*\*+/\s*$)
```
> **`Java, JavaScript` 等语言的注释内容忽略正则表达式**
>
> *这个正则表达式能够保留注释和文档内容中的 `/` 和 `*`，但需要插件在v2.5.1及以上的版本才能使用（#380），效果如下图所示：*
>
> ![2](https://user-images.githubusercontent.com/10737066/67548304-14c00780-f734-11e9-9523-f6116e27cdc4.png)

**Comment by YiiGuxing** [comment]
```regexp
\{@(code|link) (?=[^\{\}]+\})|(?<=(\{@(code|link) [^\{\}]+))\}
```
> **`Java` 文档注释中的 `{@code ...}` `{@link ...}`**
>
> *该正则表达式能够忽略掉 `{@code ...}` 或者 `{@link ...}` 中的 `{@code` ， `{@link`  和 `}`，并保留中间的内容。效果如下：*
>
> *`Instances of the class {@code Class}.` => `Instances of the class Class.`*

**Comment by 282857484** [comment]
### **直接改成默认的得了，要不还得过来翻帖子###**

**Comment by 282857484** [comment]
感谢作者大佬！
